### PR TITLE
Disable aggregate pushdown for decimal type

### LINKED ISSUE #11290: SelectiveDecimalColumnReader::scanSpec_::valueHook_ may not be nullptr
## Bug description
Exception is caused by `SelectiveDecimalColumnReader::scanSpec_::valueHook_ != nullptr` when the physical type of decimal is `int64_t` and aggregation push-down is enabled.
```
Error Source: RUNTIME
Error Code: INVALID_STATE
Retriable: False
Expression: !scanSpec_->valueHook()
Context: Split [Hive: viewfs://hadoop-meituan/ghann03/warehouse/app_hotel.db/app_poi_leopard_coverage_medal/datekey=20240910/000000_0 0 - 63768834] Task Gluten_Stage_0_TID_31_VTID_0
Additional Context: Operator: PartialAggregation[1] 1
Function: read
File: /opt/meituan/panning/gluten-dev/velox/velox/dwio/dwrf/reader/SelectiveDecimalColumnReader.cpp
Line: 115
facebook::velox::process::StackTrace::StackTrace(int)
_ZN8facebook5velox14VeloxException5State4makeIZNS1_C4EPKcmS5_St17basic_string_viewIcSt11char_traitsIcEES9_S9_S9_bNS1_4TypeES9_EUlRT_E_EESt10shared_ptrIKS2_ESA_SB_
facebook::velox::VeloxException::VeloxException(char const*, unsigned long, char const*, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, bool, facebook::velox::VeloxException::Type, std::basic_string_view<char, std::char_traits<char> >)
facebook::velox::VeloxRuntimeError::VeloxRuntimeError(char const*, unsigned long, char const*, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, bool, std::basic_string_view<char, std::char_traits<char> >)
void facebook::velox::detail::veloxCheckFail<facebook::velox::VeloxRuntimeError, facebook::velox::detail::CompileTimeEmptyString>(facebook::velox::detail::VeloxCheckFailArgs const&, facebook::velox::detail::CompileTimeEmptyString)
facebook::velox::dwrf::SelectiveDecimalColumnReader<long>::read(int, folly::Range<int const*>, unsigned long const*)
facebook::velox::dwio::common::ColumnLoader::loadInternal(folly::Range<int const*>, facebook::velox::ValueHook*, int, std::shared_ptr<facebook::velox::BaseVector>*)
facebook::velox::VectorLoader::load(folly::Range<int const*>, facebook::velox::ValueHook*, int, std::shared_ptr<facebook::velox::BaseVector>*)
facebook::velox::LazyVector::load(folly::Range<int const*>, facebook::velox::ValueHook*) const
void facebook::velox::functions::aggregate::SimpleNumericAggregate<long, long, long>::pushdown<facebook::velox::aggregate::MinMaxHook<long, false> >(char**, facebook::velox::SelectivityVector const&, std::shared_ptr<facebook::velox::BaseVector> const&)
facebook::velox::functions::aggregate::(anonymous namespace)::MaxAggregate<long>::addRawInput(char**, facebook::velox::SelectivityVector const&, std::vector<std::shared_ptr<facebook::velox::BaseVector>, std::allocator<std::shared_ptr<facebook::velox::BaseVector> > > const&, bool)
facebook::velox::exec::AggregateCompanionFunctionBase::addRawInput(char**, facebook::velox::SelectivityVector const&, std::vector<std::shared_ptr<facebook::velox::BaseVector>, std::allocator<std::shared_ptr<facebook::velox::BaseVector> > > const&, bool)
facebook::velox::exec::GroupingSet::addInputForActiveRows(std::shared_ptr<facebook::velox::RowVector> const&, bool)
facebook::velox::exec::GroupingSet::addInput(std::shared_ptr<facebook::velox::RowVector> const&, bool)
facebook::velox::exec::HashAggregation::addInput(std::shared_ptr<facebook::velox::RowVector>)
facebook::velox::exec::Driver::runInternal(std::shared_ptr<facebook::velox::exec::Driver>&, std::shared_ptr<facebook::velox::exec::BlockingState>&, std::shared_ptr<facebook::velox::RowVector>&)
facebook::velox::exec::Driver::next(std::shared_ptr<facebook::velox::exec::BlockingState>&)
facebook::velox::exec::Task::next(folly::SemiFuture<folly::Unit>*)
gluten::WholeStageResultIterator::next()
gluten::ResultIterator::getNext()
gluten::ResultIterator::hasNext()
```

I see that currently only [int128_t](https://github.com/facebookincubator/velox/blob/main/velox/functions/lib/aggregates/SimpleNumericAggregate.h#L39) won't push-down to `SelectiveDecimalColumnReader`. Perhaps, we should let `int64_t` do it as well.

### System information

Velox System Info v0.0.2
Commit: 33928e00
CMake Version: 3.29.6
System: Linux-3.10.0-862.mt20190308.130.el7.x86_64
Arch: x86_64
C++ Compiler: /opt/rh/devtoolset-10/root/usr/bin/c++
C++ Compiler Version: 10.2.1
C Compiler: /opt/rh/devtoolset-10/root/usr/bin/cc
C Compiler Version: 10.2.1

### Relevant logs

_No response_

### PR DESCRIPTION
**Pull request by NEUpanning**
Currently, `int64_t` enables push-down for decimal type. This PR disables aggregate pushdown for decimal type regardless of c++ type.

Fixes #11290

### CODE REVIEW

File: velox/functions/lib/aggregates/SimpleNumericAggregate.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ class SimpleNumericAggregate : public exec::Aggregate {
 
  protected:
   template <typename T>
-  static constexpr bool kMayPushdown = !std::is_same_v<T, int128_t> &&
```
Comment (Yuhta) [inline velox/functions/lib/aggregates/SimpleNumericAggregate.h:39]: If `T=int128_t` we for sure cannot push it down

File: velox/functions/lib/aggregates/MinMaxAggregateBase.cpp [hunk 2]
```diff
@@ -111,10 +111,14 @@ class MaxAggregate : public MinMaxAggregate<T> {
       const std::vector<VectorPtr>& args,
       bool mayPushdown) override {
     if constexpr (BaseAggregate::template kMayPushdown<T>) {
-      if (mayPushdown && args[0]->isLazy()) {
-        BaseAggregate::template pushdown<
-            velox::aggregate::MinMaxHook<T, false>>(groups, rows, args[0]);
-        return;
+      if (!args[0]->type()->isDecimal()) {
```
Comment (Yuhta) [inline velox/functions/lib/aggregates/MinMaxAggregateBase.cpp:114]:
```c++
mayPushdown && args[0]->isLazy()) && !args[0]->type()->isDecimal()
```

File: velox/functions/lib/aggregates/MinMaxAggregateBase.cpp [hunk 3]
```diff
@@ -208,10 +212,14 @@ class MinAggregate : public MinMaxAggregate<T> {
       const std::vector<VectorPtr>& args,
       bool mayPushdown) override {
     if constexpr (BaseAggregate::template kMayPushdown<T>) {
-      if (mayPushdown && args[0]->isLazy()) {
-        BaseAggregate::template pushdown<velox::aggregate::MinMaxHook<T, true>>(
-            groups, rows, args[0]);
-        return;
+      if (!args[0]->type()->isDecimal()) {
```
Comment (Yuhta) [inline velox/functions/lib/aggregates/MinMaxAggregateBase.cpp:215]: ditto

File: velox/functions/lib/aggregates/SimpleNumericAggregate.h [hunk 4]
```diff
@@ -100,19 +100,21 @@ class SimpleNumericAggregate : public exec::Aggregate {
...
     if constexpr (kMayPushdown<TData>) {
-      if (encoding == VectorEncoding::Simple::LAZY) {
-        velox::aggregate::SimpleCallableHook<TData, UpdateSingleValue> hook(
-            exec::Aggregate::offset_,
-            exec::Aggregate::nullByte_,
-            exec::Aggregate::nullMask_,
-            groups,
-            &this->exec::Aggregate::numNulls_,
-            updateSingleValue);
+      if (!arg->type()->isDecimal()) {
```
Comment (Yuhta) [inline velox/functions/lib/aggregates/SimpleNumericAggregate.h:103]: ditto

**Review by Yuhta — COMMENTED** [review]
Can you add a unit test to make sure short decimal aggregation is not push down?

**Comment by NEUpanning** [comment]
@Yuhta I've updated code based on comments and added unit test. Could you please take a look?

File: velox/exec/tests/TableScanTest.cpp [hunk 5]
```diff
@@ -3696,6 +3696,29 @@ TEST_F(TableScanTest, aggregationPushdown) {
...
+  auto rowType = ROW({"c0", "c1"}, {BIGINT(), DECIMAL(18, 2)});
+  auto vectors = makeVectors(10, 1'000, rowType);
+  auto filePath = TempFilePath::create();
+  writeToFile(filePath->getPath(), vectors);
+  createDuckDbTable(vectors);
+  auto op = PlanBuilder()
+                .tableScan(rowType)
+                .singleAggregation({"c0"}, {"min(c1)", "max(c1)", "sum(c1)"})
+                .planNode();
+  auto task = assertQuery(
```
Comment (NEUpanning) [inline velox/exec/tests/TableScanTest.cpp:3709]: @Yuhta This assert can't pass. Should I exclude decimal type from duckdb? Could you give me some guidance? Thanks!

**Comment by NEUpanning** [comment]
@Yuhta Just a gentle ping.

**Comment by Yuhta** [comment]
@NEUpanning The test failure seems related: `MinMaxTest.maxShortDecimal`

**Comment by NEUpanning** [comment]
@Yuhta For MinAggregate and MaxAggregate we should set `mayPushdown = false` if type is decimal. Otherwise, the unloaded lazyvector could cause test failures when pushdown is disabled by decimal type in `SimpleNumericAggregate::updateGroups`. I've updated code to resovle it.

**Comment by NEUpanning** [comment]
@Yuhta Could you please help to review this PR again? Thanks.

**Review by Yuhta — APPROVED** [review]

**Comment by NEUpanning** [comment]
@kevinwilfong Sure, I've rebased it onto the main branch.

**Pull request closed without merging**